core/main/src/broker/endpoint_broker.rs

```diff
@@ -421,97 +478,107 @@ impl BrokerOutputForwarder {
         tokio::spawn(async move {
             while let Some(mut v) = rx.recv().await {
                 let mut is_event = false;
+                // First validate the id check if it could be an event
                 let id = if let Some(e) = v.get_event() {
                     is_event = true;
                     Some(e)
                 } else {
                     v.data.id
                 };
+
                 if let Some(id) = id {
                     if let Ok(broker_request) = platform_state.endpoint_state.get_request(id) {
                         let sub_processed = broker_request.is_subscription_processed();
-                        let rpc_request = broker_request.rpc;
+                        let rpc_request = broker_request.rpc.clone();
                         let session_id = rpc_request.ctx.get_id();
                         let is_subscription = rpc_request.is_subscription();
-                        if let Some(session) = platform_state
-                            .session_state
-                            .get_session_for_connection_id(&session_id)
-                        {
-                            let request_id = rpc_request.ctx.call_id;
-                            v.data.id = Some(request_id);
-                            if let Some(result) = v.data.result.clone() {
-                                if is_event {
-                                    if let Some(filter) = broker_request
-                                        .rule
-                                        .transform
-                                        .get_filter(super::rules_engine::RuleTransformType::Event)
-                                    {
-                                        if let Ok(r) = jq_compile(
-                                            result,
-                                            &filter,
-                                            format!("{}_event", rpc_request.ctx.method),
-                                        ) {
+
+                        // Step 1: Create the data
```

Comment:
For more readable and maintainable code, it may be be helpful to move the different steps to separate functions.

Reply:
This method was already refactored challenge was sharing the variables across these refactored methods which made it very. Did little bit more refactoring anything more would increase the parameter complexity of these methods